### ISSUE TITLE
Support for Pi Zero (W)?

### ISSUE THREAD

**Issue by zachariahfuller** [pihnat/rpi-Candle#10]
Am I an idiot for trying to run Candle on a Pi Zero W? It's all I've got at this point what with supply of Pis being what it is. I recognize that the Pi Zero W uses the original RPi ARM v6 processor, so there's probably something I have to do to get it to run, if it does run.

I've tried building Denvi's Candle and am getting errors in the make process, but I don't know if this is because I am missing dependencies or there's an actual error in the code:

> frmmain.cpp: In member function 'void frmMain::preloadSettings()':
> frmmain.cpp:324:5 error: 'QGLFormat' was not declared in this scope
> 324 | QGLFormat fmt = QGLFormat::defaultFormat();
> frmmain.cpp:325:5: error: 'fmt' was not declared in this scope; did you mean 'fma'?
> 325 | fmt.setSwapInterval(set.value("vsync", false).toBool() ? 1 : 0);

I attempted to install the following packages to build this from the source, but my RasPi already had the latest versions installed:
libqt5core5a libqt5gui5 libqt5serialport5 libqt5serialport5-dev libqt5svg5 libqt5svg5-dev libqt5widgets5 libqt5charts5-dev libqt5charts5 libudev-dev libusb-1.0-0-dev

Note that qt5-default is not available for the ARM v6 architecture.

Any help figuring out what's wrong would be greatly appreciated here! I'm hoping it's something stupid, and not a "this was never going to work" scenario. :)

**Comment by zachariahfuller** [comment]
It turns out I _am_ an idiot, but not for the reason I thought! pihnat's rpi-Candle will totally build and install correctly on the Zero (W) (or early Pis) due to the changes made from Denvi's Candle. This appears to be agnostic of the ARM architecture version.

I'm actually encouraged with how well it runs on the Zero W; I was expecting a train wreck due to lack of resources. Then, I have yet to actually process an .nc file... It's not the snappiest application, but it's about as fast as anything on the Zero W. I'll report when I've processed (or at least simulated) some .nc code.

**Comment by pihnat** [comment]
Hi
Great news, I was going to ask if you followed my wiki because there were some changes required.

But if it's anything like running it on a raspi3B+ then it'll be sluggish and frustrating. You'll be surprised at how responsive it is on a raspi4 - just like using a PC, I wouldn't recommend anything less than a raspi4. I believe raspi 400s are available so maybe that might be the way to go for the moment until supplies of raspi4 increase.

Thanks for the info about running on a Zero - I had never thought of that but at least we now know it's possible.
Peter

**Comment by zachariahfuller** [comment]
I was able to simulate an NC file I made some time ago. On my laptop simulation took approximately 1 eye blink; on the Pi Zero W, it took about 2:14. 😁 I have yet to actually run a job but I'm really quite pleased with how well it runs, opens files, etc. It can be a little sluggish, but not frustratingly so. I imagine a Pi 4 would do much better, but this was hardware I had lying around.
One thing I noticed is that the spindle speed slider only goes to 100, which my router interprets as 100_rpm_... In my original Denvi Candle installation, the max is 10k, which works just fine. Was this intentional? Can I change the maximum slider value?
